readthedocs/proxito/views/mixins.py

```diff
@@ -352,34 +351,34 @@ def canonical_redirect(
 
         :param request: Request object.
         :param final_project: The current project being served.
-        :param version_slug: The current version slug being served.
-        :param filename: The filename being served.
         :param redirect_type: The type of canonical redirect (https, canonical-cname, subproject-main-domain)
-        :param external: If the version is from a pull request preview.
+        :param external_version_slug: The version slug if the request is from a pull request preview.
```

Comment:
Do we need to pass this in, or can we pull it off the request?

Reply:
We can get it from the unresolved_domain from the request, but it felt more explicit not having these methods depend on the request having all that.